transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -666,11 +666,10 @@ private void processSelectedKey(SelectionKey k, AbstractNioChannel ch) {
             // and thus the SelectionKey could be cancelled as part of the deregistration process, but the channel is
             // still healthy and should not be closed.
             // See https://github.com/netty/netty/issues/5125
-            if (eventLoop != this || eventLoop == null) {
-                return;
+            if (eventLoop == this) {
+                // close the channel if the key is not valid anymore
+                unsafe.close(unsafe.voidPromise());
             }
-            // close the channel if the key is not valid anymore
-            unsafe.close(unsafe.voidPromise());
             return;
```

Comment:
this `return` can be removed now.

Reply:
this seems to violate the original logic. The healthy channel  may be closed when `eventLoop != this`

Reply:
@MrYangxf not sure I understand your concern... With your change `return` is the last statement in the method which makes it redundant.. What I am missing ?

Reply:
@normanmaurer  ...`return` is contained in the `if(!k.isValid())` statement instead of the last line of the method...

Reply:
@MrYangxf doh! stupid me ... I did not notice that there is another scope. You are right, of course.